code/game/gamemodes/meteor/meteor.dm

```diff
@@ -11,17 +12,20 @@
 	world << "<B>The space station has been stuck in a major meteor shower. You must escape from the station or at least live.</B>"
 
 
-/datum/game_mode/meteor/post_setup()
-//	defer_powernet_rebuild = 2//Might help with the lag
-	spawn(meteordelay)
-		nometeors = 0
-	..()
-
-
 /datum/game_mode/meteor/process()
-	if(nometeors) return
-
-	spawn() spawn_meteors(1, meteors_normal)
+	if(nometeors || meteordelay > world.time - round_start_time) 
+		return
+	
+	var/list/wavetype = meteors_normal
+	var/meteorminutes = (world.time - round_start_time - meteordelay) / 10 / 60
+	
+	
+	if (prob(meteorminutes/2))
+		wavetype = meteors_catastrophic
+	else if (prob(meteorminutes))
+		wavetype = meteors_threatening

```

Comment:
Switch the order of these two, and remove the `else`. It'll give you more accurate results as `meteorminutes` increases.